packages/x-date-pickers/src/ClockPicker/ClockPicker.tsx

```diff
@@ -267,44 +266,63 @@ export const ClockPicker = React.forwardRef(function ClockPicker<TDate extends u
         return false;
       }
 
-      const validateTimeValue = (
-        value: number,
-        getRequestedTimePoint: (when: 'start' | 'end') => TDate,
-      ) => {
-        const isAfterComparingFn = createIsAfterIgnoreDatePart(
-          disableIgnoringDatePartForTimeValidation,
-          utils,
-        );
-
-        return Boolean(
-          (minTime && isAfterComparingFn(minTime, getRequestedTimePoint('end'))) ||
-            (maxTime && isAfterComparingFn(getRequestedTimePoint('start'), maxTime)) ||
-            (shouldDisableTime && shouldDisableTime(value, viewType)),
-        );
+      const isAfter = createIsAfterIgnoreDatePart(disableIgnoringDatePartForTimeValidation, utils);
+
+      const shouldDisableDatePeriod = ({
+        start,
+        end,
+        value,
+      }: {
```

Comment:
Doing the two verifications at the same time is disturbing

Could be split into two distinct functions:
- `containsValidTime({start, end})`
- `isValidTime(value)` 

Then `return !containsValidTime({start, end}) || !isValidTime(value)`

Reply:
Done :+1: